packages/ai/src/ui/chat.ts

```diff
@@ -416,17 +427,19 @@ export abstract class AbstractChat<UI_MESSAGE extends UIMessage> {
               : part,
           );
       }
-    });
-  };
 
-  /**
-   * Checks if the assistant message can be submitted, i.e. if it
-   * has tool calls and all tool calls have results.
-   *
-   * @returns {boolean} True if the assistant message can be submitted, false otherwise.
-   */
-  canAssistantMessageBeSubmitted = (): boolean =>
-    isAssistantMessageWithCompletedToolCalls(this.lastMessage);
+      // automatically send the message if the sendAutomaticallyWhen function returns true
+      if (
+        this.status !== 'streaming' &&
+        this.status !== 'submitted' &&
+        this.sendAutomaticallyWhen?.({ messages })
```

Comment:
@lgrammel was just trying this out, and noticed that when `this.sendAutomaticallyWhen` is called here, `messages` doesn't contain the updates tool result messages. should this be `this.state.messages` instead?

Reply:
just submitted a PR here https://github.com/vercel/ai/pull/7545